package-parser/tests/data/optionals/annotation_data.json

```diff
@@ -8,5 +8,10 @@
         "defaultType": "string",
         "defaultValue": "test2",
         "target": "test/test/some_global_function/optional_parameter_that_should_be_optional_with_different_default"
+    },
+    "test/test/some_global_function/required_parameter_that_should_be_optional_while_ignoring_the_non_string": {
```

Comment:
Wrong logic: In this case an optional annotation should **not** be added since we don't know what the true most common value is. For instance, the variable `non_string` might have the value `'test'` all 10 times it was used.

Reply:
Scratch that, I like your version to filter out all non-literal values better now. Otherwise, we would almost never be able to add `@optional` annotations, since there will always be some non-literal values, even if they are not the most common one.